src/components/clusterWizard/ClusterDetailsForm.tsx

```diff
@@ -0,0 +1,125 @@
+import React from 'react';
+import _ from 'lodash';
+import { Formik, FormikHelpers } from 'formik';
+import { Cluster, ClusterCreateParams, ClusterUpdateParams, ManagedDomain } from '../../api/types';
+import ClusterWizardStep from './ClusterWizardStep';
+import { Grid, GridItem } from '@patternfly/react-core';
+import { canNextClusterDetails } from './wizardTransition';
+import { OpenshiftVersionOptionType } from '../../types/versions';
+import ClusterWizardStepHeader from './ClusterWizardStepHeader';
+import ClusterWizardFooter from './ClusterWizardFooter';
+import { getFormikErrorFields } from '../ui/formik/utils';
+import ClusterDetailsFormFields from './ClusterDetailsFormFields';
+import { ClusterDetailsValues } from './types';
+import { getClusterDetailsInitialValues, getClusterDetailsValidationSchema } from './utils';
+
+type ClusterDetailsFormProps = {
+  cluster?: Cluster;
+  pullSecret: string;
+  managedDomains: ManagedDomain[];
+  ocpVersions: OpenshiftVersionOptionType[];
+  usedClusterNames: string[];
+
+  navigation: React.ReactNode;
+
+  moveNext: () => void;
+  handleClusterCreate: (params: ClusterCreateParams) => Promise<void>;
+  handleClusterUpdate: (clusterId: Cluster['id'], params: ClusterUpdateParams) => Promise<void>;
+};
+
+const ClusterDetailsForm: React.FC<ClusterDetailsFormProps> = (props) => {
```

Comment:
I hope I do not miss something. The `ClusterDetetailsForm` is used by the `ClusterDetails` (Create cluster wizard flow). Honestly, there is no more need for that split and we could leave content of the `ClusterDetailsForm` insinde of the `ClusterDetails` but it happend to be that way because of the second round of refactorings ... Anyway, the benefit of that split can be in having 2 smaller components instead of a single bigger one.

Reply:
Right, I agree that splitting is fine. My bad, GitHub did not load the diff of `ClusterDetails.tsx` and I thought that it is being removed.